lib/migration.c

```diff
@@ -289,7 +289,7 @@ handle_data_offset(vfu_ctx_t *vfu_ctx, struct migration *migr,
         if (is_write) {
             /* TODO writing to read-only registers should be simply ignored */
             vfu_log(vfu_ctx, LOG_ERR, "bad write to migration data_offset");
-            return -EINVAL;
+            return ERROR_INT(EINVAL);
         }
         ret = migr->callbacks.prepare_data(vfu_ctx, offset, NULL);
```

Comment:
This is a good opportunity to update `prepare_data` documentation specifing that -1 must be returned on failure and errno must be set.

Reply:
yep thanks